shared/src/commonMain/kotlin/org/hyperskill/app/manage_subscription/presentation/ManageSubscriptionReducer.kt

```diff
@@ -50,12 +55,36 @@ internal class ManageSubscriptionReducer : StateReducer<State, Message, Action>
     private fun fetchSubscription(): ReducerResult =
         State.Loading to setOf(InternalAction.FetchSubscription)
 
-    private fun handleManageSubscriptionClicked(state: State): ReducerResult =
-        if (state is State.Content && state.manageSubscriptionUrl != null) {
-            state to setOf(
-                InternalAction.LogAnalyticsEvent(ManageSubscriptionClickedManageHyperskillAnalyticEvent),
-                Action.ViewAction.OpenUrl(state.manageSubscriptionUrl)
-            )
+    private fun handleActionButtonClicked(state: State): ReducerResult =
+        if (state is State.Content) {
+            state to when {
+                state.subscription.isActive && state.manageSubscriptionUrl != null -> {
```

Comment:
`ManageSubscriptionViewStateMapper` has the same condition, you can extract it to state extension